pylint/checkers/base.py

```diff
@@ -298,28 +298,40 @@ def _get_break_loop_node(break_node):
 
 def _loop_exits_early(loop):
     """
-    Returns true if a loop may end with a break statement.
+    Returns true if a loop may end with a break or return statement.
 
     Args:
         loop (astroid.For, astroid.While): the loop node inspected.
 
     Returns:
-        bool: True if the loop may end with a break statement, False otherwise.
+        bool: True if the loop may end with a break or return statement, False otherwise.
     """
     loop_nodes = (nodes.For, nodes.While)
     definition_nodes = (nodes.FunctionDef, nodes.ClassDef)
+    break_nodes = (nodes.Break, nodes.Return)
     inner_loop_nodes = [
         _node
         for _node in loop.nodes_of_class(loop_nodes, skip_klass=definition_nodes)
         if _node != loop
     ]
     return any(
         _node
-        for _node in loop.nodes_of_class(nodes.Break, skip_klass=definition_nodes)
+        for _node in loop.nodes_of_class(break_nodes, skip_klass=definition_nodes)
         if _get_break_loop_node(_node) not in inner_loop_nodes
+        and not _node_in_orelse(loop, _node)
     )
 
 
+def _node_in_orelse(loop, node):
```

Comment:
```suggestion
def _node_in_orelse(loop:Union[nodes.While, nodes.For], node: Union[nodes.Break, nodes.Return]) -> bool:
```

Perhaps even add a docstring? I like those even for internal methods, but I know opinions differ.

Reply:
I'm also for docstring if there is more information than in the function name and argument names themselves. Here if the docstring was """Is the node in an or else inside the given loop ?""" I think it's the same info that function name and arg name, only duplicated.